app/controllers/application_controller.rb

```diff
@@ -10,4 +10,10 @@ class ApplicationController < ActionController::Base
   def record_not_found
     render plain: "404 Not Found", status: 404
   end
+
+  def call_rake(task, options = {})
+    options[:rails_env] ||= Rails.env
+    args = options.map { |n, v| "#{n.to_s.upcase}='#{v}'" }
+    system "rake #{task} #{args.join(' ')} --trace 2>&1 >> #{Rails.root}/log/rake.log &"
```

Comment:
I think Rake has a facility to call tasks within its lib -- avoid system calls if possible

Reply:
@govuklaurence - the reason why as system call is being used is so that the process can be run in the background. I have not been able to find a way to do this with Rake::Task - any ideas?